c10/core/impl/TorchDispatchModeTLS.h

```diff
@@ -21,5 +21,17 @@ struct C10_API TorchDispatchModeTLS {
 
 C10_API bool dispatch_mode_enabled();
 
+struct C10_API DisableTorchDispatchModeGuard {
+  DisableTorchDispatchModeGuard() : prior_(TorchDispatchModeTLS::get_state()) {
+    TorchDispatchModeTLS::set_state(TorchDispatchModeTLS());
+  }
+  ~DisableTorchDispatchModeGuard() {
+    TorchDispatchModeTLS::set_state(std::move(prior_));
+  }
+
+ private:
+  TorchDispatchModeTLS prior_;
+};
```

Comment:
Hmm, this seems to overlap w/ StashTorchDispatchModeGuard, except yours takes in a prior_

Can we combine them?

Reply:
Didn't see that class, will merge.